src/invidious/database/migrations/0008_create_playlists_table.cr

```diff
@@ -0,0 +1,48 @@
+module Invidious::Database::Migrations
+  class CreatePlaylistsTable < Migration
+    version 8
+
+    def up(conn : DB::Connection)
+      if !privacy_type_exists?(conn)
+        conn.exec <<-SQL
+        CREATE TYPE public.privacy AS ENUM
+        (
+          'Public',
+          'Unlisted',
+          'Private'
+        );
+        SQL
+      end
+
+      conn.exec <<-SQL
+      CREATE TABLE IF NOT EXISTS public.playlists
+      (
+        title text,
+        id text primary key,
+        author text,
+        description text,
+        video_count integer,
+        created timestamptz,
+        updated timestamptz,
+        privacy privacy,
+        index int8[]
+      );
+      SQL
+
+      conn.exec <<-SQL
+      GRANT ALL ON public.playlists TO current_user;
+      SQL
+    end
+
+    private def privacy_type_exists?(conn : DB::Connection) : Bool
```

Comment:
Instead of the weird pgsql query, I extracted out this check to look for the privacy type and only creating it if this method returns false.